### DISCUSSION TITLE
400 when trying to override configuration of API Loader

### DISCUSSION THREAD

**Discussion by poniatowski1987** [FlowiseAI/Flowise#1576] [Q&A] [answered]
Hi Folks,
newbie Flowise user here. The issue I'm facing might be trivial - perhaps there's something obvious I'm missing. In short, Flowise returns 400 when I try to override API Loader configuration in my call, and I am unable to see why. Thanks in advance for your help on this. More details below.

My use case is a universal PDF chatbot allowing to specify the question AND the URL to a PDF in the payload.

I've built a simple Railway flow where I load the PDF through an API Loader node calling APIfy with "PDF Text Extractor" actor exposed (https://apify.com/jirimoravcik/pdf-text-extractor). APIfy endpoint takes the PDF URL in the payload.

It all works beautifully if I hardcode the PDF URL by clicking on "Additional Parameters" in GUI, where I can add any field to the body of the APIfy call.
<img width="896" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/7233f391-f3b0-4c7d-9b5d-d0fdf24c17b8">
Both the build-in chatbot and Flowise API work properly. I can see Flowise calls APIfy, the hardcoded URL is passed in the body; APIfy then returns the PDF, and Flowise does the rest of its magic.
APIfy receives:
<img width="508" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/c9ad2d09-6c6f-471e-866e-cdc8d16ac3cf">
Flowise returns:
<img width="913" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/13e43aae-e829-48bb-8365-d52d63ecd097">

**But**: I don't want the PDF URL to be hardcoded - I want to specify it each time when calling Flowise. To do it, I need to be able to override the API Loader node configuration when doing so.

I start off by removing the hard-coded "Additional Parameters" in API Loader node, and then follow Flowise's instructions to put what I need inside the body. The options for API Loader are:
<img width="824" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/afa026dd-8cca-4431-87bd-5f917a6feeca">
("URL" is the URL of the endpoint API Loader is to call.)

Now, I've tried multiple approaches to syntax (it's unclear to me from instructions), but it seems no matter which one I try, I get a 400 from Flowise. I can see Flowise never calls APIfy. Is there something I'm missing?

<img width="723" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/3a7a0015-77a2-48b4-8010-3673010408dc">

<img width="732" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/68443026/78e9cb95-8875-4ffe-b634-67edb4f582a9">

**Comment by HenryHengZJ** [comment, marked as answer]
@poniatowski1987 dont you have to put the urls in the body?
```
overrideConfig: {
    url: "apify-url",
    body: {
        urls: ["xxxx.pdf"]
    }
}
```

**Reply by poniatowski1987** [reply]
@HenryHengZJ You're a lifesaver. It worked perfectly. I thought I had to put the node name in the payload. Thanks so much for your help!
